### PR TITLE
Integrate remaining items from SDK feedback

### LINKED ISSUE #171: Some questions regarding `nucypher-ts` API/SDK
I did a very cursory review of the `nucypher-ts` code base and had some questions regarding the design of the API. Please excuse my ignorance and bear with me regarding any misconceptions about ts/js common practices:

1. Any reason `Operator.OR` and `Operator.AND` can't be static singletons?
2. How come supported chains ids declared in Condition base class?
3. How come `EvmCondition` doesn't extend `RPCCondition`?
4. Is `function_abi` supported in nucypher-ts? Related to nucypher/tdec#99 
     * EvmCondition doesn't seem that general, but rather specific methods? Related to `function_abi` support?
5. Does `getContextParameters` check for context variables in key - which is currently supported in the python-side (although I have questions about that - see https://github.com/nucypher/nucypher/issues/3027#issuecomment-1325209266
6. `createWalletSignature`, should probably use `cbd` instead of `tdec` for the `name` property in `domain` for the EIP signature data.

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
- Closes #171
- [x] Q1: Addressed in a [different PR](https://github.com/nucypher/nucypher-ts/pull/200/files#diff-1959e640158d146fb12ebd41a71133a2cf0e4167e80f88ab723bb9fe991ab351R59)
- [x] Q2: Done in this PR
- [ ] Q3: I will address that in a separate PR - I want to refactor conditions structure anyway
- [x] Q4: Done in https://github.com/nucypher/tdec/issues/99
- [x] Q5: Done in https://github.com/nucypher/nucypher-ts/pull/199
- [x] Q6: Done in this PR

### CODE REVIEW

File: src/conditions/providers.ts [hunk 1]
```diff
@@ -66,7 +66,7 @@ export class WalletAuthenticationProvider {
         ],
       },
       domain: {
-        name: 'tDec',
+        name: 'cbd',
```
Comment (piotr-roslaniec) [inline src/conditions/providers.ts:69]: Q6

File: src/agents/contracts.ts [hunk 2]
```diff
@@ -20,10 +20,16 @@ const GOERLI: Contracts = {
...
+const MAINNET: Contracts = {
+  SUBSCRIPTION_MANAGER: undefined,
+  COORDINATOR: 'undefined',
+};
+
 const CONTRACTS: { readonly [key in ChainId]: Contracts } = {
   [ChainId.POLYGON]: POLYGON,
   [ChainId.MUMBAI]: MUMBAI,
   [ChainId.GOERLI]: GOERLI,
+  [ChainId.MAINNET]: MAINNET,
```
Comment (derekpierre) [inline src/agents/contracts.ts:32]: Side question (just for my understanding): how come MAINNET wasn't defined previously?

**Review by derekpierre — APPROVED** [review]
🎸

File: src/agents/contracts.ts [hunk 3]
```diff
@@ -20,10 +20,16 @@ const GOERLI: Contracts = {
   COORDINATOR: '0x2cf19429168a0943992D8e7dE534E9b802C687B6',
 };
 
+const MAINNET: Contracts = {
+  SUBSCRIPTION_MANAGER: undefined,
+  COORDINATOR: 'undefined',
```
[the remark this answers is not in the document]
Comment (piotr-roslaniec) [reply src/agents/contracts.ts:25]: That's a mistake right there

**Pull request merged**